validator_worker/src/main.rs

```diff
@@ -202,23 +202,25 @@ async fn validator_tick<A: Adapter + 'static>(
     let duration = Duration::from_millis(config.validator_tick_timeout as u64);
 
     match channel.spec.validators.find(&whoami) {
-        SpecValidator::Leader(_) => match timeout(duration, leader::tick(&sentry)).await {
-            Err(e) => return Err(ValidatorWorkerError::Failed(e.to_string())),
-            Ok(Err(e)) => return Err(ValidatorWorkerError::Failed(e.to_string())),
-            _ => (),
-        },
-        SpecValidator::Follower(_) => match timeout(duration, follower::tick(&sentry)).await {
-            Err(e) => return Err(ValidatorWorkerError::Failed(e.to_string())),
-            Ok(Err(e)) => return Err(ValidatorWorkerError::Failed(e.to_string())),
-            _ => (),
-        },
-        SpecValidator::None => {
-            return Err(ValidatorWorkerError::Failed(
-                "validatorTick: processing a channel where we are not validating".to_string(),
-            ))
-        }
-    };
-    Ok(())
+        SpecValidator::Leader(_) => timeout(duration, leader::tick(&sentry))
```

Comment:
yeah, I am not sure this better than the previous implementation. With the previous implementation it's very easy to understand that a Result is embedded in another Result but with this refactor it's not clear at all

Reply:
Moved it back to `match`